src/components/transaction-dialog/transaction-dialog.js

```diff
@@ -0,0 +1,72 @@
+import angular from "angular";
+import template from "./transaction-dialog.html";
+
+import ngAnimate from "angular-animate";
+import ngMaterial from "angular-material";
+
+export class TransactionDialogController {
+  constructor($mdDialog, $filter) {
+    this.$mdDialog = $mdDialog;
+    this.$filter = $filter;
+    this.tags = ["transport", "food", "salary"];
+    this.accounts = ["Privat", "BoaBank", "Cash"];
+  }
+
+  $onChanges(changes) {
+    if (changes.isOpen.currentValue) {
+      this.showDialog(changes.isOpen.currentValue.event);
+    }
+  }
+
+  showDialog(ev) {
+    this.$mdDialog.show({
+      template,
+      controllerAs: "$ctrl",
+      controller: DialogController,
+      parent: angular.element(document.body),
+      targetEvent: ev,
+      clickOutsideToClose: true,
+      locals: {
+        $mdDialog: this.$mdDialog,
```

Comment:
isn't `$mdDialog` injected without this?

Reply:
no

Reply:
ok

Reply:
yes. tried another way, than last time